packages/liveblocks-react-ui/src/components/internal/Attachment.tsx

```diff
@@ -0,0 +1,306 @@
+import React, { memo, useCallback, useEffect, useMemo, useState } from "react";
+
+import { CrossIcon } from "../../icons/Cross";
+import { SpinnerIcon } from "../../icons/Spinner";
+import { WarningIcon } from "../../icons/Warning";
+import type { GlobalOverrides } from "../../overrides";
+import { useOverrides } from "../../overrides";
+import { AttachmentTooLargeError } from "../../primitives";
+import { useComposerAttachmentsContextOrNull } from "../../primitives/Composer/contexts";
+import { classNames } from "../../utils/class-names";
+import { formatFileSize } from "../../utils/format-file-size";
+import { Tooltip } from "./Tooltip";
+
+interface FileAttachmentProps extends ComponentPropsWithoutRef<"div"> {
+  attachment: CommentMixedAttachment;
+  onDeleteClick?: MouseEventHandler<HTMLButtonElement>;
+  preventFocusOnDelete?: boolean;
+  overrides?: Partial<GlobalOverrides>;
+}
+
+const IMAGE_PREVIEW_MAX_SIZE = 50 * 1024 * 1024; // 50 MB
+
+const fileExtensionRegex = /^(.+?)(\.[^.]+)?$/;
+
+function splitFileName(name: string) {
+  const match = name.match(fileExtensionRegex);
+
+  return { base: match?.[1] ?? name, extension: match?.[2] };
+}
+
+function getFileAttachmentIconGlyph(mimeType: string) {
+  if (
+    mimeType === "application/zip" ||
+    mimeType === "application/x-rar-compressed" ||
+    mimeType === "application/x-7z-compressed" ||
+    mimeType === "application/x-zip-compressed" ||
+    mimeType === "application/x-tar" ||
+    mimeType === "application/gzip"
+  ) {
+    return (
+      <path d="M13 15h2v1h-1.5a.5.5 0 0 0 0 1H15v1h-1.5a.5.5 0 0 0 0 1H15v1h-1.5a.5.5 0 0 0 0 1h1a.5.5 0 0 0 .5-.5V20h1.5a.5.5 0 0 0 0-1H15v-1h1.5a.5.5 0 0 0 0-1H15v-1h1.5a.5.5 0 0 0 .5-.5V15a2 2 0 0 1 2 2v4a2 2 0 0 1-2 2h-4a2 2 0 0 1-2-2v-4a2 2 0 0 1 2-2Z" />
+    );
+  }
+
+  if (mimeType.startsWith("text/")) {
+    return (
+      <path d="M10 16a.5.5 0 0 1 .5-.5h5a.5.5 0 0 1 0 1h-5a.5.5 0 0 1-.5-.5Zm0 2a.5.5 0 0 1 .5-.5h9a.5.5 0 0 1 0 1h-9a.5.5 0 0 1-.5-.5Zm0 2a.5.5 0 0 1 .5-.5h7a.5.5 0 0 1 0 1h-7a.5.5 0 0 1-.5-.5Zm0 2a.5.5 0 0 1 .5-.5h8a.5.5 0 0 1 0 1h-8a.5.5 0 0 1-.5-.5Z" />
+    );
+  }
+
+  if (mimeType.startsWith("image/")) {
+    return (
+      <path d="M12 16h6a1 1 0 0 1 1 1v3l-1.293-1.293a1 1 0 0 0-1.414 0L14.09 20.91l-.464-.386a1 1 0 0 0-1.265-.013l-1.231.985A.995.995 0 0 1 11 21v-4a1 1 0 0 1 1-1Zm-2 1a2 2 0 0 1 2-2h6a2 2 0 0 1 2 2v4a2 2 0 0 1-2 2h-6a2 2 0 0 1-2-2v-4Zm3 2a1 1 0 1 0 0-2 1 1 0 0 0 0 2Z" />
+    );
+  }
+
+  if (mimeType.startsWith("video/")) {
+    return (
+      <path d="M12 15.71a1 1 0 0 1 1.49-.872l4.96 2.79a1 1 0 0 1 0 1.744l-4.96 2.79A1 1 0 0 1 12 21.29v-5.58Z" />
+    );
+  }
+
+  if (mimeType.startsWith("audio/")) {
+    return (
+      <path d="M15 15a.5.5 0 0 0-.5.5v7a.5.5 0 0 0 1 0v-7a.5.5 0 0 0-.5-.5Zm-2.5 2.5a.5.5 0 0 1 1 0v3a.5.5 0 0 1-1 0v-3Zm-2 1a.5.5 0 0 1 1 0v1a.5.5 0 0 1-1 0v-1Zm6-1a.5.5 0 0 1 1 0v3a.5.5 0 0 1-1 0v-3ZM19 16a.5.5 0 0 0-.5.5v5a.5.5 0 0 0 1 0v-5a.5.5 0 0 0-.5-.5Z" />
+    );
+  }
+
```

Comment:
Could add a `startsWith("font/") && "application/vnd.ms-fontobject"` icon if we're being really picky!

Reply:
I'm adding them to the "text" icon list for now 👍